signac/contrib/schema.py

```diff
@@ -46,48 +97,97 @@ def remove_dict_placeholder(x):
 
 
 class ProjectSchema(object):
-    "A description of a project's state point schema."
+    """A description of a project's state point schema.
+
+    Parameters
+    ----------
+    schema : dict
+        Project schema.
 
+    """
     def __init__(self, schema=None):
         if schema is None:
             schema = dict()
         self._schema = schema
 
     @classmethod
     def detect(cls, statepoint_index):
+        """Detect Project's state point schema.
+
+        Parameters
+        ----------
+        statepoint_index :
+            State point index.
+
+        Returns
+        -------
+        type : `signac.contrib.schema.ProjectSchema`
+            The detected project schema.
+
+        """
         return cls({k: _collect_by_type(v) for k, v in statepoint_index})
 
     def format(self, depth=None, precision=None, max_num_range=None):
         """Format the schema for printing.
 
-        :param depth:
+        Parameters
+        ----------
+        depth : int
             A non-zero value will return a nested formatting up to the specified depth,
             defaults to 0.
-        :type depth:
-            int
-        :param precision:
-            Round numerical values up the give precision, defaults to unlimited precision.
-        :type precision:
-            int
-        :param max_num_range:
+        precision : int
+            Round numerical values to the given precision, defaults to unlimited precision.
+        max_num_range : int
             The maximum number of entries shown for a value range, defaults to 5.
-        :type max_num_range:
-            int
-        :returns:
+
+        Returns
+        -------
+        str
             A formatted representation of the project schema.
+
         """
         if depth is None:
             depth = 0
         if max_num_range is None:
             max_num_range = 5
 
         def _fmt_value(x):
+            """Convert a value to a string, rounded to a given precision.
+
+            Parameters
+            ----------
+            x : float or int
```

Comment:
The inputs don't have to be numerical, they could be strings. I suggest leaving no type specifier here.
```suggestion
            x :
```